### PR TITLE
fix bugs

### PR DESCRIPTION
**Pull request by cqc-melf**

### CODE REVIEW

File: pytket/qir/conversion/api.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ def pytket_to_qir(
     :type pyqir_0_6_compatibility: bool
     """
 
-    if len(circ.q_registers) > 1 or circ.q_registers[0].name != "q":
+    if len(circ.q_registers) > 1 or (
```
Comment (cqc-melf) [inline pytket/qir/conversion/api.py:58]: This should solve all problems we had with circuits without qubits.

**Comment by cqc-melf** [comment]
Solves the following bugs:
* phaseop throws an error
* copy bits only working on registers
* set bits only working on registers
* bit shift not working

File: pytket/qir/conversion/conversion.py [hunk 2]
```diff
@@ -855,9 +847,47 @@ def condition_block() -> None:
...
-                for out in outputs:
-                    self.set_cregs[out] = command.op.values
+                assert len(command.op.values) == len(command.bits)
+                assert len(command.qubits) == 0
+
+                for b, v in zip(command.bits, command.op.values):
+                    if v:
+                        output_instruction = pyqir.const(self.qir_bool_type, 1)
+                    else:
+                        output_instruction = pyqir.const(self.qir_bool_type, 0)
```
Comment (cqc-alec) [inline pytket/qir/conversion/conversion.py]:
```suggestion
                    output_instruction = pyqir.const(self.qir_bool_type, v)
```
Does this work?

File: pytket/qir/conversion/conversion.py [hunk 3]
```diff
@@ -855,9 +847,47 @@ def condition_block() -> None:
...
+                            self.ssa_vars[b.reg_name],
+                            pyqir.const(self.qir_int_type, b.index[0]),
+                            output_instruction,
+                        ],
+                    )
+
+            elif isinstance(op, CopyBitsOp):
+                assert len(command.qubits) == 0
+                half_length = int(len(command.args) / 2)
+                assert 2 * half_length == len(command.args)
```
Comment (cqc-alec) [inline pytket/qir/conversion/conversion.py]:
```suggestion
                assert len(command.args) % 2 == 0
                half_length = len(command.args) // 2
```
Equivalent but clearer imo.

File: tests/conversion_test.py [hunk 4]
```diff
@@ -152,6 +155,221 @@ def test_pytket_qir_7() -> None:
...
+def test_pytket_qir_9() -> None:
+    # test copybits op
+    c = Circuit(1, name="test_classical")
+    a = c.add_c_register("a", 2)
+    b = c.add_c_register("b", 2)
+
+    c.add_c_copyreg(a, b)
+
+    assert c.n_qubits == 1
+    assert c.n_bits == 4
```
Comment (cqc-alec) [inline tests/conversion_test.py]: Small point, but I don't think these asserts (and similar ones below) are relevant here; they're not testing pytket-qir at all.

File: pytket/qir/conversion/conversion.py (hunk 3 above)
Comment (cqc-melf) [reply pytket/qir/conversion/conversion.py]: Done in @cqc-melf
[suggested changes: CopyBits](https://github.com/CQCL/pytket-qir/pull/45/commits/de8d6e3272fa6ae482bcf28f5753a8087a55ddf8)
[de8d6e3](https://github.com/CQCL/pytket-qir/pull/45/commits/de8d6e3272fa6ae482bcf28f5753a8087a55ddf8)

File: pytket/qir/conversion/conversion.py (hunk 2 above)
Comment (cqc-melf) [reply pytket/qir/conversion/conversion.py]: No, but it looks like int(v) is working. 

Done in @cqc-melf
[suggested changes: SetBitsOp](https://github.com/CQCL/pytket-qir/pull/45/commits/ed0bbd93526504f38535f7dd66aa84454adbb2da)
[ed0bbd9](https://github.com/CQCL/pytket-qir/pull/45/commits/ed0bbd93526504f38535f7dd66aa84454adbb2da)

File: tests/conversion_test.py (hunk 4 above)
Comment (cqc-melf) [reply tests/conversion_test.py]: I have cleaned that up, done in @cqc-melf
[remove qubit and bit asserts](https://github.com/CQCL/pytket-qir/pull/45/commits/ace002f6e8fceb14e15e00e30e1874296fc3eb2b)
[ace002f](https://github.com/CQCL/pytket-qir/pull/45/commits/ace002f6e8fceb14e15e00e30e1874296fc3eb2b)

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**